src/main/java/hudson/tasks/junit/JUnitStep.java

```diff
@@ -0,0 +1,185 @@
+package hudson.tasks.junit;
+
+import hudson.*;
+import hudson.model.Result;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.remoting.VirtualChannel;
+import org.apache.tools.ant.DirectoryScanner;
+import org.apache.tools.ant.types.FileSet;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepDescriptorImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractSynchronousStepExecution;
+import org.jenkinsci.plugins.workflow.steps.StepContextParameter;
+import org.jenkinsci.remoting.RoleChecker;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.logging.Logger;
+
+/**
+ * Workflow step that archives JUnit Test results and returns a summary of counts
+ * to the caller
+ */
+public class JUnitStep extends AbstractStepImpl {
+
+    private static final Logger LOGGER = Logger.getLogger(JUnitStep.class.getName());
+
+    @Nullable
+    private String location;
+
+    private boolean keepLongStdio;
+
+    @DataBoundConstructor
+    public JUnitStep(@Nullable String location) {
+        this.location = location;
+    }
+
+    @Nullable
+    public String getLocation() {
+        return location;
+    }
+
+    public boolean isKeepLongStdio() {
+        return keepLongStdio;
+    }
+
+    @DataBoundSetter
+    public void setLocation(@Nullable String location) {
+        this.location = location;
+    }
+
+    public void setKeepLongStdio(boolean keepLongStdio) {
+        this.keepLongStdio = keepLongStdio;
+    }
+
+    public static final class ExecutionImpl extends AbstractSynchronousStepExecution<TestResultSummary> {
+
+        @Inject
+        private transient JUnitStep step;
+
+
+        @StepContextParameter
+        private transient TaskListener listener;
+
+        @StepContextParameter
+        private transient FilePath workspace;
+
+        @StepContextParameter
+        private transient EnvVars env;
+
+        @StepContextParameter
+        private transient Run build;
+
+
+        @Override
+        protected TestResultSummary run() throws Exception {
+
+            final long buildTime = build.getTimestamp().getTimeInMillis();
+            final long timeOnMaster = System.currentTimeMillis();
+
+            String location = step.location;
+            if (location == null) {
+                location = "**/target/surefire-reports/TEST-*.xml";
```

Comment:
Better just make this field mandatory, and if you like specify the `default` attribute on `f:textbox` in `config.jelly` (and/or suggest this value in `help-location.html`).